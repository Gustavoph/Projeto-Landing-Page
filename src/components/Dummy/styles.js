import styled, { css } from 'styled-components';

export const Dummy = styled.div`
  ${({ theme }) => css``}
`;
