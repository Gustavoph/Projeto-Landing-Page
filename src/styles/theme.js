export const theme = {
  colors: {
    main: '#cdcdcd',
    second: '#fdfd',
  },
};
