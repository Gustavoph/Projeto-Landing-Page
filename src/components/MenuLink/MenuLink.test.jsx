import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { MenuLink } from '.';

describe('<MenuLink />', () => {
  it('should render a link', () => {
    renderTheme(<MenuLink link="https://www.google.com/">Children</MenuLink>);
    expect(screen.getByRole('link', { name: 'Children' })).toHaveAttribute(
      'target',
      '_self',
    );
  });

  it('should render a link', () => {
    renderTheme(
      <MenuLink link="https://www.google.com/" newTab={true}>
        Children
      </MenuLink>,
    );
    expect(screen.getByRole('link', { name: 'Children' })).toHaveAttribute(
      'target',
      '_blank',
    );
  });

  it('should render a link', () => {
    const { container } = renderTheme(
      <MenuLink link="https://www.google.com/" newTab={false}>
        Children
      </MenuLink>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
