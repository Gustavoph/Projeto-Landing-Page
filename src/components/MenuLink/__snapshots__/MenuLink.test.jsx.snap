// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should render a link 1`] = `
.c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  width: 0;
  bottom: 0.75rem;
  left: 50%;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  width: 50%;
  left: 25%;
}

<a
  class="c0"
  href="https://www.google.com/"
  target="_self"
>
  Children
</a>
`;
